app/javascript/menu/second-level.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { SideNavItems, SideNavMenu, SideNavMenuItem } from 'carbon-components-react/es/components/UIShell';
+import { itemId, linkProps } from './item-type';
+
+const mapItems = (items) => items.map((item) => (
+  item.items.length
```

Comment:
What about pulling out the component selection logic outside from the render?
```jsx
const Component = item.items.length ? MenuSection : MenuItem;
return <Component key={item.id} {...item} />
```

Reply:
nah, first level section/item take different props